Sources/JS/printTypes.js

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 for (var varNumber in types) {
```

Comment:
I think here you can use `Object.keys` instead of the `for-in` plus `if`. Also, that function should probably be backed up the same way `getObjectPropertyNames` is. Finally, please make a short comment here to explain why we can't just use a for-in loop (because we must ignore properties on the prototype that script code may have set).

Reply:
hmm I don't quite understand how you want to iterate without that **if**, because the additional property is on object prototype -> you encounter this property when iterating any array too (e.g. result of Object.keys too)

Reply:
As far as I understand, Object.keys will only return you the names of own properties and ignore properties on the prototype: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys with that it should work, shouldn't it?

Reply:
yes as we discussed on call if we iterate over indeces not array itself it will work